lib/filter-cloudwatch-compiler.js

```diff
@@ -68,80 +78,187 @@ var FilterCloudWatchCompiler = ASTVisitor.extend({
         }
     },
 
+    // Build up a list of conditions that control what metrics to
+    // fetch. Each item in the array specifies conditions on:
+    //  - product: an AWS product.
+    //  - item: a list of item ids.
+    //  - metric: a list of metrics to fetch.
+    //
+    // While parsing the filter expression any of these can not yet
+    // defined (undefined for product, an empty array for
+    // items/metrics).
+    //
+    // ANDs are used to combine incomplete conditions and ORs are used
+    // to add to the list of conditions.
+
     visitBinaryExpression: function(node) {
         var left, right, ret;
 
         switch (node.operator) {
 
-            case 'OR':
+            case 'AND':
                 left = this.visit(node.left);
                 right = this.visit(node.right);
 
-                ret = _.extend({}, right, left);
+                // You can only combine single conditions
+                if (left.length > 1 || right.length > 1) {
+                    this.throwUnsupportedFilter(node.location, 'AND between anything other than simple conditions');
+                }
+                left = left[0];
+                right = right[0];
+
+                // You can only combine items if one of the products/items/metrics
+                // is wildcarded (undefined for products, an empty array for items/metrics).
+                if (left.product !== undefined &&
+                    right.product !== undefined) {
+                    this.throwUnsupportedFilter(node.location, 'AND between products');
+                }
+
+                if ((left.item.length > 0 && right.item.length > 0)) {
+                    this.throwUnsupportedFilter(node.location, 'AND between items');
+                }
+
+                if ((left.metric.length > 0 && right.metric.length > 0)) {
+                    this.throwUnsupportedFilter(node.location, 'AND between metrics');
+                }
```

Comment:
Consider this instead of the above if statements:
```
_.each(['product', 'item', 'metric'], (type) => {
  if ((left[type].length > 0 && right[type].length > 0)) {
    this.throwUnsupportedFilter(node.location, `AND between ${type}s`);
  }
})
```